src/vivarium_public_health/population/base_population.py

```diff
@@ -235,16 +235,56 @@ def _load_population_structure(self, builder: Builder) -> pd.DataFrame:
             raise ValueError(
                 f"Scaling factor must be a pandas DataFrame. Provided value: {scaling_factor}"
             )
-        scaling_factor = scaling_factor.set_index(
-            [col for col in scaling_factor.columns if col != "value"]
+        formatted_pop_structure, formatted_scaling_factor = self._format_data_inputs(
+            population_structure, scaling_factor
         )
-        population_structure = population_structure.set_index(
-            [col for col in population_structure.columns if col != "value"]
-        )
-        scaled_population_structure = (population_structure * scaling_factor).reset_index()
+
+        scaled_population_structure = (
+            formatted_pop_structure * formatted_scaling_factor
+        ).reset_index()
 
         return scaled_population_structure
 
+    def _format_data_inputs(
+        self, pop_structure: pd.DataFrame, scalar_data: pd.DataFrame
+    ) -> tuple[pd.DataFrame, pd.DataFrame]:
+        """Data cleaning function to check whether scalar_data and population structure are compatible for scaling
+        the population structure of a simulation. This method will do any necessary transformations to
```

Comment:
Can you expand this a bit? I really don't understand why this function exists based on this docstring.